schemas/json/.wp-env.json

```
@@ -0,0 +1,92 @@
+{
+	"title": "JSON schema for WordPress wp-env configuration files",
+	"$schema": "http://json-schema.org/draft-04/schema#",
+	"definitions": {
+		"//": {
+			"reference": "https://developer.wordpress.org/block-editor/reference-guides/packages/packages-env/"
+		},
+		"wpEnvProperties": {
+			"properties": {
+				"core": {
+					"description": "The WordPress installation to use. If null is specified, wp-env will use the latest production release of WordPress.",
+					"default": "null",
+					"oneOf": [
+						{
+							"type": "string"
+						},
+						{
+							"type": "null"
+						}
+					]
+				},
+				"phpVersion": {
+					"description": "The PHP version to use. If null is specified, wp-env will use the default version used with production release of WordPress.",
+					"default": "null",
+					"oneOf": [
+						{
+							"type": "string"
+						},
+						{
+							"type": "null"
+						}
+					]
+				},
+				"plugins": {
+					"description": "A list of plugins to install and activate in the environment.",
+					"default": [],
+					"type": "array",
+					"items": {
+						"type": "string"
+					}
+				},
+				"themes": {
+					"description": "A list of themes to install in the environment.",
+					"default": [],
+					"type": "array",
+					"items": {
+						"type": "string"
+					}
+				},
+				"port": {
+					"description": "The primary port number to use for the installation. You'll access the instance through the port: 'http://localhost:8888'.",
+					"default": 8888,
+					"type": "integer"
+				},
+				"config": {
+					"description": "Mapping of wp-config.php constants to their desired values.",
+					"default": {},
+					"type": "object"
+				},
+				"mappings": {
+					"description": "Mapping of WordPress directories to local directories to be mounted in the WordPress instance.",
+					"default": {},
+					"type": "object"
+				}
+			}
+		}
+	},
+	"type": "object",
+	"anyOf": [
```

```suggestion
	"allOf": [
```

JSON schema validates each of the subschemas independently of one another.
- `allOf` is like a logical AND where _all_ of the subschemas need to match.
- `anyOf` is like a logical OR where _at least one_ of the subschemas needs to match.
- `oneOf` is like a logical XOR where _exactly one_ of the subschemas needs to match.

So when you want to combine properties, you want the nested properties AND the additional top-level properties to both be validated with `allOf`. First the `$ref` will be tested, then if that passes, the top-level properties subschema will be tested on its own.

(It's also why you need to add another entire subschema for testing `"additionalProperties": false` below.)